src/utils/components/CapacityInput/CapacityInput.tsx

```diff
@@ -24,37 +23,27 @@ type CapacityInputProps = {
 
 const CapacityInput: React.FC<CapacityInputProps> = ({ size, onChange, label }) => {
   const { t } = useKubevirtTranslation();
-  const [selectOpen, toggleSelect] = React.useState(false);
+  const [selectOpen, toggleSelect] = useState<boolean>(false);
+  const [unit, setUnit] = useState<string>();
+  const [value, setValue] = useState<number>(0);
 
-  const [value, quantityUnit] = bytesFromQuantity(size);
+  useEffect(() => {
+    if (size) {
+      const [unitValue] = size.match(/[a-zA-Z]+/g);
+      const [sizeValue = 0] = size.match(/[0-9]+/g) || [];
+      setUnit(!unitValue?.endsWith('B') ? unitValue + 'B' : unitValue);
+      setValue(+sizeValue);
+    }
+  }, [size]);
```

Comment:
```suggestion
  const [unitValue] = size?.match(/[a-zA-Z]+/g);
  const [sizeValue = 0] = size?.match(/[0-9]+/g) || [];
  const unit = !unitValue?.endsWith('B') ? unitValue + 'B' : unitValue;
  const value = +sizeValue;
```

What do you think about?

Reply:
It is better! Thank you :)